### PR TITLE
[4.0] Move changelog JS to it's own file

### PR DESCRIPTION
**Pull request by roland-d**
Pull Request for Issue #24849

Here goes my first try at some modular JS

### Summary of Changes
Fixed the missing modal IDs in the link
Moved all changelog related JavaScript to it's own file

### Testing Instructions
1. Apply this pull request
2. Run npm run build:js
3. Install the test extension from https://github.com/roland-d/component_joomla/releases/tag/v0.0.2
4. Go to Manage
5. Filter the list on List of Students
6. Click on the version number 0.0.2
7. A modal shows up with the changelog as shown below under Manage View
8. Go to Update
9. Click on Find Updates
10. The update is shown and click on the Changelog button
11. A modal shows up with the changelog as shown below under Update View

### Expected result
There is no JavaScript error in Update view
Specific changelog JavaScript is used

### Actual result
There is a JavaScript error in Update view
Core Joomla JavaScript is used

### Documentation Changes Required
None

Please review:
@euismod2336 @C-Lodder

### CODE REVIEW

**Review by wilsonge — APPROVED** [review]
Looks good to me.

File: build/media_source/com_installer/js/changelog.es6.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+   *             this is used to determine which version number to show
+   *
+   * @since   4.0.0
+   */
+  Joomla.loadChangelog = (extensionId, view) => {
+    Joomla.request({
+      url: `index.php?option=com_installer&task=manage.loadChangelog&eid=${extensionId}&source=${view}&format=json`,
+      onSuccess: (response) => {
+        const result = JSON.parse(response);
+        document.querySelectorAll(`#changelogModal${extensionId} .modal-body`)[0].innerHTML = result.data;
```
Comment (C-Lodder) [inline build/media_source/com_installer/js/changelog.es6.js]: No need to use `querySelectorAll` here. Just do:

```
document.querySelector(`#changelogModal${extensionId} .modal-body`).innerHTML = result.data;
```

**Review by C-Lodder — CHANGES_REQUESTED** [review]

File: build/media_source/com_installer/js/changelog.es6.js (hunk 1 above)
Comment (roland-d) [reply build/media_source/com_installer/js/changelog.es6.js]: Thanks. Done.

File: build/media_source/com_installer/js/changelog.es6.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+   * @param view The view the changelog is for,
+   *             this is used to determine which version number to show
+   *
+   * @since   4.0.0
+   */
+  Joomla.loadChangelog = (extensionId, view) => {
+    Joomla.request({
+      url: `index.php?option=com_installer&task=manage.loadChangelog&eid=${extensionId}&source=${view}&format=json`,
+      onSuccess: (response) => {
+        const result = JSON.parse(response);
```
Comment (euismod2336) [inline build/media_source/com_installer/js/changelog.es6.js]: Please add some error handling, to check if the response is what we expect it to be

File: build/media_source/com_installer/js/changelog.es6.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        document.querySelector(`#changelogModal${extensionId} .modal-body`).innerHTML = result.data;
+      },
+    });
```
Comment (euismod2336) [inline build/media_source/com_installer/js/changelog.es6.js:53]: We don't need handling in case of status code != 200?

**Comment by ghazal** [comment]
I have tested this item :white_check_mark: successfully on 50951c0a76357925169ea21a11dfeb04fd0bfc06<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/24890">issues.joomla.org/tracker/joomla-cms/24890</a>.</sub>

issue-24890

File: build/media_source/com_installer/js/changelog.es6.js (hunk 2 above)
Comment (roland-d) [reply build/media_source/com_installer/js/changelog.es6.js]: Done

File: build/media_source/com_installer/js/changelog.es6.js (hunk 3 above)
Comment (roland-d) [reply build/media_source/com_installer/js/changelog.es6.js:53]: Done

**Comment by euismod2336** [comment]
I have tested this item :white_check_mark: successfully on 3672c3ac4d59f2d23d843491143f6ac464d5f02f<br /><br />works as advertised<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/24890">issues.joomla.org/tracker/joomla-cms/24890</a>.</sub>

**Comment by euismod2336** [comment]
thnx @roland-d

**Comment by roland-d** [comment]
@ghazal Could you please test again? Thank you.

**Comment by ghazal** [comment]
I have tested this item :white_check_mark: successfully on c31d4af67e814f6595650e745578c893f449c459<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/24890">issues.joomla.org/tracker/joomla-cms/24890</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/24890">issues.joomla.org/tracker/joomla-cms/24890</a>.</sub>

**Comment by roland-d** [comment]
pinging @wilsonge for merge

**Comment by HLeithner** [comment]
thx

**Comment by roland-d** [comment]
Thank you

**Pull request merged**